### ISSUE TITLE
Intent ref not working

### ISSUE THREAD

**Issue by arnaud-moncel** [openNLX/javascript#78]
When using https://github.com/openNLX/javascript/blob/e26637481d5b54a4796f4e6a54f31596b7b342d7/src/openNLX.js#L167 function the intent ref feature not working.

To fix it add :
```js
if (!intentsProvider && env.agent) {
      intentsProvider = env.agent.getIntentsProvider(env.version);
      e.intentsProvider = intentsProvider;
}
```
on https://github.com/openNLX/javascript/blob/e26637481d5b54a4796f4e6a54f31596b7b342d7/src/nlx/resolver.js#L201

**Comment by mikbry** [comment]
Need more infos, resolver.ref.test.js tests this feature

**Comment by jeffladiray** [comment]
It just doesn't work online when `{{#id}}` is used. IntentProvider is manually provided in ` resolver.ref.test.js` test cases. But `OpenNLX.parse` doesn't provide use a way to specify `intentProvider` from the exposed API.

@arnaud-moncel's code address this issue by providing intentProvider as a part of env, which fix the issue we've faced on our code

**Comment by mikbry** [comment]
FIXED in v0.8.6
